assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx

```diff
@@ -1,17 +1,30 @@
 /* eslint-disable react/no-array-index-key */
-import React, { useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 
 import { groupBy } from '@lib/lists';
 
 import { getCatalog } from '@state/selectors/catalog';
 import { updateCatalog } from '@state/actions/catalog';
+import {
+  providerData,
+  getLabels,
+  getProviderByLabel,
+} from '../ProviderLabel/ProviderLabel';
 import CatalogContainer from './CatalogContainer';
 import CheckItem from './CheckItem';
+import ProviderSelection from './ProviderSelection';
+
+const ALL_FILTER = 'All';
 
 // eslint-disable-next-line import/prefer-default-export
 export function ChecksCatalogNew() {
   const dispatch = useDispatch();
+  const updatedProvider = providerData;
```

Comment:
Hey @EMaksy ,
I have looked at this, and we cannot do this direct assignment, as this simply copies the reference of the variable, so changing `updatedProvider` changes `providerData` afterwards. In order to avoid this we need to do a shallow copy, like this: `const updatedProviderData = {...providerData, '': ALL_FILTER}`.
This shallow copies the variable and adds the new empty field to the new object.

After that, we need to change where `providerData` is used by `updatedProviderData` all around the code in this file

Reply:
Hey good point, i applied the  suggested changes. Thanks